### PR TITLE
Add Caffeine exclusion for errorprone and checker

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Refactoring

### Description

TL;DR: these dependencies are superfluous and are not need at runtime.  There's more explanation why caffeine has these at compile scope here:  https://github.com/ben-manes/caffeine/issues/474#issuecomment-728243942.

No functional impact.

_Please describe your pull request_

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -43,7 +43,7 @@
         <kafka.version>3.6.1</kafka.version>
         <kroxy.extension.version>0.8.1</kroxy.extension.version>
         <log4j.version>2.22.1</log4j.version>
-        <caffeine.version>3.1.8</caffeine.version>
+        <caffeine.version>3.1.8.temporary-redhat-00001</caffeine.version>
```
Comment (franvila) [inline pom.xml:46]: Is this version existing in maven repository?

Comment (k-wall) [reply pom.xml:46]: sorry - that was unintentional.

**Review by franvila — APPROVED** [review]
LGTM
